trailblazer/apps/slurm/models.py

```diff
@@ -0,0 +1,72 @@
+"""Model SLURM output."""
+from datetime import datetime
+from typing import List, Dict, Optional, Any
+
+from pydantic import BaseModel, Field, validator
+
+from trailblazer.constants import SlurmJobStatus, SlurmSqueueHeader
+from trailblazer.utils.datetime import convert_days_to_min, convert_timestamp_to_min
+
+
+class SqueueJob(BaseModel):
+    """Model job meta data from squeue output."""
+
+    id: int = Field(..., alias=SlurmSqueueHeader.JOBID.value)
+    step: str = Field(..., alias=SlurmSqueueHeader.NAME.value)
+    status: SlurmJobStatus = Field(..., alias=SlurmSqueueHeader.STATE.value)
+    time_limit: str = Field(..., alias=SlurmSqueueHeader.TIME_LIMIT.value)
+    time_elapsed: int = Field(..., alias=SlurmSqueueHeader.TIME.value)
+    started_at: Optional[datetime] = Field(..., alias=SlurmSqueueHeader.START_TIME.value)
+
+    @validator("status", always=True, pre=True)
+    def convert_status_to_lower_case(cls, value: str) -> str:
+        """Convert string to lower case."""
+        return value.lower()
+
+    @validator("time_elapsed", always=True, pre=True)
+    def convert_time_elapsed_to_min(cls, value: str) -> int:
```

Comment:
this function seems to validate an input, determine a time format and convert time. Can the separate aspects be extracted into smaller functions?

Reply:
Fixed